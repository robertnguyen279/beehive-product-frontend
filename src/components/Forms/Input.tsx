import React from 'react';
import { InputType } from 'types/main';

const Input = ({
  type,
  value,
  className,
  name,
  onChange,
  prefixIcon,
  style,
  error,
  ...config
}: InputType): React.ReactElement => {
  return (
    <div className={`form__input${error ? '--error' : ''} relative ${className ? className : ''}`} style={{ ...style }}>
      <input
        type={type}
        value={value}
        name={name}
        onChange={onChange}
        {...config}
        className={`bg-gray-50 ${prefixIcon ? 'pr-5 pl-12' : 'px-5'} font-xl w-full h-10`}
      />
      <span className="relative text-sm text-red-600">{error}</span>

      {prefixIcon && (
        <div className="form__icon-prefix h-full absolute bg-white flex items-center justify-center">
          <img src={prefixIcon} alt="input-icon" className="w-1/2" />
        </div>
      )}
    </div>
  );
};

export default Input;
