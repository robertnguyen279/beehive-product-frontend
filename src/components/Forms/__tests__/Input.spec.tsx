import React from 'react';
import { shallow } from 'enzyme';

import Input from '../Input';
// const clickFn = jest.fn();

describe('Input', () => {
  const component = shallow(
    <Input
      type="text"
      value="test"
      name="test"
      error="test"
      onChange={(e) => {
        console.log(e);
      }}
    />,
  );

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
