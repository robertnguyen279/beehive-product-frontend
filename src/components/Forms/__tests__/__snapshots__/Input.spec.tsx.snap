// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div
  className="form__input--error relative "
  style={Object {}}
>
  <input
    className="bg-gray-50 px-5 font-xl w-full h-10"
    name="test"
    onChange={[Function]}
    type="text"
    value="test"
  />
  <span
    className="relative text-sm text-red-600"
  >
    test
  </span>
</div>
`;
